src/middlewared/middlewared/etc_files/local/smb4.conf

```diff
@@ -0,0 +1,330 @@
+            pc.update({'strict locking': 'No'})
+            pc.update({'deadtime': '15'})
+            pc.update({'max log size': '51200'})
+            pc.update({'load printers': 'No'})
+            pc.update({'printing': 'bsd'})
+            pc.update({'disable spoolss': 'Yes'})
+            if db['cifs']['cifs_srv_guest'] is not "nobody":
+                pc.update({'guest account': db['cifs']['cifs_srv_guest']}) 
+            pc.update({'map to guest': 'Bad User'})
+            if db['cifs']['cifs_srv_obey_pam_restrictions']:
+                pc.update({'obey pam restrictions': 'Yes'}) 
+            if db['cifs']['cifs_srv_ntlmv1_auth']:
+                pc.update({'ntlm auth': 'Yes'})
+                pc.update({'client ntlmv2 auth': 'No'})
+            pc.update({'directory name cache size': '0'})
+            pc.update({'kernel change notify': 'No'})
+            pc.update({'panic action': "/usr/local/libexec/samba/samba-backtrace"})
+            pc.update({'nsupdate command': "/usr/local/bin/samba-nsupdate -g"})
+            if db['cifs']['cifs_srv_description']:
+                pc.update({'server string': db['cifs']['cifs_srv_description']})
+            pc.update({'lm announce': 'Yes'})
+            if db['cifs']['cifs_srv_hostlookup']:
+                pc.update({'hostname lookups': 'Yes'})
+            if not db['cifs']['cifs_srv_unixext']:
+                pc.update({'unix extensions': 'No'})
+            if db['cifs']['cifs_srv_timeserver']:
+                pc.update({'time server': 'Yes'})
+            if db['cifs']['cifs_srv_nullpw']:
+                pc.update({'null passwords': 'Yes'})
+            if db['cifs']['cifs_srv_allow_execute_always']:
+                pc.update({'acl allow execute always': 'Yes'})
+            pc.update({'dos filemode': 'Yes'})
+            if not db['cifs']['cifs_srv_zeroconf']:
+                pc.update({'multicast dns register': 'No'})
+            if db['cifs']['cifs_srv_filemask']:
+                pc.update({'create mask': db['cifs']['cifs_srv_filemask']})
+            if db['cifs']['cifs_srv_dirmask']:
+                pc.update({'directory mask': db['cifs']['cifs_srv_dirmask']})
+            pc.update({'dos charset': db['cifs']['cifs_srv_doscharset']})
+            pc.update({'unix charset': db['cifs']['cifs_srv_unixcharset']})
+
+            return pc 
+
+        def add_licensebased_params(pc, db):
+            if not db['truenas_conf']['smb_ha_mode']:
+                pc.update({'netbios name': db['cifs']['cifs_srv_netbiosname']})
+                pc.update({'netbios aliases': db['cifs']['cifs_srv_netbiosalias']})
+            if db['truenas_conf']['smb_ha_mode'] == "legacy":
+                failover_node = middleware.call_sync('notifier.failover_node')
+                if db['truenas_conf']['failover_status'] == "MASTER":
+                    pc.update({'netbios aliases': db['cifs']['cifs_srv_netbiosalias']})
+                if failover_node == "A":
```

Comment:
If we use sharing.smb.config `netbiosname` will return the correct value for each node.